### PR TITLE
Change from netcoreapp2.1 to netcoreapp3.1 and also add net6.0 version

### PR DESCRIPTION
**Pull request by satbai**
Adding netcoreapp3.1 as the default netcore version but also building net6.0 version so we can separately release that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +71,8 @@ Examples:
 
 Using the above is recommended, but as per [NuGet's plugin discovery rules](https://github.com/NuGet/Home/wiki/NuGet-cross-plat-authentication-plugin#plugin-installation-and-discovery), alternatively you can install the credential provider to a location you prefer, and then set the environment variable NUGET_PLUGIN_PATHS to the .dll of the credential provider found in plugins\netcore\CredentialProvider.Microsoft\CredentialProvider.Microsoft.dll. For example, $env:NUGET_PLUGIN_PATHS="my-alternative-location\CredentialProvider.Microsoft.dll".
 
+Users requiring .NET 6 can manually downloade the .NET 6 version `Microsoft.Net6.NuGet.CredentialProvider` of the [1.0.0 release](https://github.com/microsoft/artifacts-credprovider/releases/tag/v1.0.0).
```
Comment (edgarrs) [inline README.md:74]: typo download

Comment (embetten) [reply README.md:74]: @edgarrs once the PMs sign off let's complete this PR with the typo and I will create a follow up to fix after it's been merged.

**Review by edgarrs — APPROVED** [review]

File: helpers/installcredprovider.ps1 [hunk 2]
```diff
@@ -80,7 +80,7 @@ if (!$releaseId) {
 $releaseUrl = [System.IO.Path]::Combine($releaseUrlBase, $releaseId)
 $releaseUrl = $releaseUrl.Replace("\","/")
 
-$zipFile = "Microsoft.NetCore2.NuGet.CredentialProvider.zip"
+$zipFile = "Microsoft.NetCore3.NuGet.CredentialProvider.zip"
```
Comment (AArnott) [inline helpers/installcredprovider.ps1:83]: This appears to have broken use of this script.
> Unable to resolve the Credential Provider zip file from https://api.github.com/repos/Microsoft/artifacts-credprovider/releases/latest

It turns out that this script looks for a file by this name in the latest release from this repo, but [the latest release](https://github.com/microsoft/artifacts-credprovider/releases/tag/v0.1.28) only has a file by the name of Microsoft.NetCore**2**.NuGet.CredentialProvider.zip.

This is breaking by cloud builds. Please either issue a new release with a Microsoft.NetCore3 file, or revert this change.

Comment (edgarrs) [reply helpers/installcredprovider.ps1:83]: Thank you @AArnott for calling it out. We reverted the change until we are ready to release the new version.

Comment (AArnott) [reply helpers/installcredprovider.ps1:83]: Confirmed. The script is working now. Thank you.

**Pull request merged**